src/monster.cpp

```diff
@@ -192,9 +192,15 @@ void monster::update_check(){
     float upgrade_chance = 1000 * (1- pow(std::max(0.0, 0.5 - type->base_upgrade_chance * .01 * elapsed_lives), elapsed_lives));
     add_msg(m_debug, "Upgrade chance: %f", upgrade_chance);
     if (upgrade_chance > rng(0, 999)){
-        const auto monsters = MonsterGroupManager::GetMonstersFromGroup(type->upgrade_group);
-        const std::string newtype = monsters[rng(0, monsters.size() - 1)];
-        poly(GetMType(newtype));
+        if (type->upgrade_group != "NULL"){
```

Comment:
I'd probably do this the other way around; have it test to see if it can upgrade into a specific monster first, then have it check for the group, but it's not that big of a deal.